### ISSUE TITLE
iOS Safari "Add to Home Screen" web-app fails to load PDF in helloworld example - status 206

### ISSUE THREAD

**Issue by rorysmorris** [mozilla/pdf.js#7377]
Configuration:
- Web browser and its version: Safari
- Operating system and its version: iOS 9.3 (latest)
- PDF.js version: latest

Steps to reproduce the problem:
1.  Add `<meta name="apple-mobile-web-app-capable" content="yes">` to `<head>` of helloworld example
2.  Replace PDF with a larger PDF
3.  View in iOS Safari, then hit Share and "Add to Home Screen"
4.  Open web-app, PDF will fail to load;
5.  Any code inside `PDFJS.getDocument(pdfUrl).then(function(pdf) {` will not run. 

This works as expected in iOS Safari. When the page is saved to the home-screen, it stops working. 

Using Safari's web inspector, in the real iOS Safari app, in the Network tab - the full PDF transfers with a status of 200 and it displays the full file-size.  When saved to the home screen, only 64kb of the PDF transfers and the status is 206.

I have tried using compatibility.js as I've read this is due to range requests, however the same problem persists.  Can anybody help me please?

**Comment by rorysmorris** [comment]
Further information:

This isn't necessarily a bug with PDF.js.  I've just tested loading a large PDF into an iframe on iOS Safari and the same problem appears:  status 200 and a full-download in Safari,  but status 206 and a 64kb download when saved to the home screen. 

Interestingly though, this works when using viewer.html.  If I add the `web-app-capable` meta tag to viewer.html, point it to a large PDF and save it to the home screen, it has no problem downloading the PDF file.  Does something exist in viewer.js that bypasses the way helloworld downloads the PDF?

**Comment by yurydelendik** [comment]
Notice that in [web/compatibility.js](https://github.com/mozilla/pdf.js/blob/master/web/compatibility.js#L456) we disable HTTP range request. Only Safari has an issue with caching of the range requests, so see if setting disableRange=true will help your case (or use entire compabitility.js)

**Comment by yurydelendik** [comment]
@rorysmorris notice that once you used range requests once they stay in cache. try to purge the cache and try using compatibility.js again.

**Comment by rorysmorris** [comment]
Thanks for your reply @yurydelendik 

I found that by forcing disableRange to true or false, by hacking the code in compatibility.js - this made no difference at all. I even reset my iPad back to factory settings to make sure it wasn't a caching issue.

I have no idea why, but the issue seemed to originate from my environment itself, when I upload the web-app I'm working on to an FTP server, the problem goes away.

I was serving up my web-app through MAMP on OSX, and accessing it on the iPad via my network IP address, i.e.  `http://192.168.0.123:8888`

Whether the problem is MAMP, my machine, my network, the iPad - I do not know.

**Comment by timvandermeij** [comment]
From the description here it sounds like this problem is not related to PDF.js itself, but more related to a specific configuration. Therefore I don't think there is anything we can do about this on the PDF.js side. Closing for now, but we can reopen this if there is more information that the problem is really caused by PDF.js.

**Comment by loveaurell** [comment]
@rorysmorris I have the same problem. It never gets loaded. I've placed a console.log in the callback of PDFJS.getDocument(srcToPdf) but it never gets called. In safari, it loads fine.

Is iOS webapps really capable of loading web workers?I've tried setting disableWorker = true; and it says 'using fake worker' but it still tries to load the worker file. Is this correct behavior? What is 'disableWorker = true' supposed to do?

**Comment by rorysmorris** [comment]
@loveaurell I managed to get this working in the end, the problem was due to the environment - when I uploaded to a proper web-server it all worked okay. 

In your Network tab, do you see the PDF downloading as only 64kb? If so that's problem the same issue I had.

**Comment by loveaurell** [comment]
@rorysmorris I can't find I way to properly debug in "add-to-homescreen" mode. I'm trying to use safari on my macbook and debug the iphone through usb but somehow the network requests are not showing up correctly in "add-to-homescreen" mode. So it's hard to say if it starts to download the pdf file at all. Any advice on how to monitor the network calls is appreciated!

**Comment by rorysmorris** [comment]
@loveaurell 

I'd suggest downloading XCode which has an iOS Simulator you can run on your MacBook. Then, when you save your app to the home screen on the Simulator, and then launch the app from your home screen, a new menu item appears in Safari (when you go to the Develop menu item at the top, you'll see Simulator > App Name).

That is how I was working anyway.

**Comment by loveaurell** [comment]
@rorysmorris 
I managed to monitor network traffic with simulator/safari and noticed that the downloaded size is indeed 64kB and it does not continue to download the other chunks. Also tried using a small 21kB file which downloaded fine. I really don't know what to do about this, since I've tried using both my own nginx-server as well as putting the pdf on dropbox and downloading from there with the same result. If the chunk size was configurable I could probably around it by setting it higher than 64kB.
